protocol/service_test.go

```diff
@@ -191,3 +177,39 @@ func checkError(err error) {
 		panic(err)
 	}
 }
+
+// ServiceMock implements protocol.Service interface implement the methods to
```

Comment:
Maybe better: "ServiceMock implements the protocol.Servce interface and the methods to be used in the test"